packages/ses/src/tame-locale-methods.js

```diff
@@ -0,0 +1,54 @@
+import { assert } from './assert.js';
+import { getOwnPropertyNames, defineProperty } from './commons.js';
+
+const localePattern = /^(\w*[a-z])[L]ocale([A-Z]\w*)$/;
+
+// See https://tc39.es/ecma262/#sec-string.prototype.localecompare
+const nonLocaleCompare = function localeCompare(that) {
+  if (this === null || this === undefined) {
+    throw new TypeError(`this is not coercible`);
+  }
+  const s = `${this}`;
+  that = `${that}`;
+  if (s < that) {
+    return -1;
+  }
+  if (s > that) {
+    return 1;
+  }
+  assert(s === that, `expected ${s} and ${that} to compare`);
+  return 0;
+};
+
+export default function tameLocaleMethods(intrinsics, localeTaming = 'safe') {
+  if (localeTaming !== 'safe' && localeTaming !== 'unsafe') {
+    throw new Error(`unrecognized dateTaming ${localeTaming}`);
+  }
+  if (localeTaming === 'unsafe') {
+    return;
+  }
+
+  defineProperty(String.prototype, 'localeCompare', {
+    value: nonLocaleCompare,
+  });
+
+  for (const intrinsicName of getOwnPropertyNames(intrinsics)) {
+    const intrinsic = intrinsics[intrinsicName];
+    for (const methodName of getOwnPropertyNames(intrinsic)) {
+      const match = localePattern.exec(methodName);
+      if (match) {
+        assert(
+          typeof intrinsic[methodName] === 'function',
+          `expected ${methodName} to be a function`,
+        );
+        const nonLocaleMethodName = `${match[1]}${match[2]}`;
+        const method = intrinsic[nonLocaleMethodName];
```

Comment:
This might break in interesting ways if the language grows a new locale method that doesn’t have a corresponding non-locale-named method. Perhaps we should use an explicit list.

Reply:
That's what the immediately following `assert` test on line 48 is about. Is it adequate?

Reply:
The assert ensures that we fail fast. Could we alternately just succeed since it would be culled?

Reply:
culled?

Reply:
If my intuition is right, any method not expressly in the list of allowed methods would be deleted by lockdown, after taming. So, if a shim or future version of the language added a method with Locale in its name and no corresponding non-locale method, we don’t need to tame it. There’s also some chance that any new non-locale method isn’t a suitable replacement.

Reply:
That all makes sense. But does it suggest any change to this PR before I merge?

Reply:
IIUC, the culling question does not suggest any change to the PR so I'll proceed to merge. Please let me know if I'm missing something.